src/trace-writer.js

```diff
@@ -62,12 +64,51 @@ function TraceWriter(logger, options) {
   /** @private {Boolean} whether the trace writer is active */
   this.isActive = true;
 
+  if (onUncaughtExceptionValues.indexOf(config.onUncaughtException) === -1) {
+    logger.error('The value of onUncaughtException should be one of ',
+      onUncaughtExceptionValues);
+    throw new Error('Invalid value for onUncaughtException configuration.');
+  }
+  var onUncaughtException = config.onUncaughtException;
+  if (onUncaughtException !== 'ignore') {
+    var that = this;
+    this.unhandledException_ = function() {
+      that.flushBuffer_();
+      if (onUncaughtException === 'flushAndExit') {
+        setTimeout(function() {
+          process.exit(1);
+        }, 2000);
+      }
+    };
+    process.on('uncaughtException', this.unhandledException_);
+  }
+}
+util.inherits(TraceWriter, common.Service);
+
+TraceWriter.prototype.stop = function() {
+  this.isActive = false;
+};
+
+TraceWriter.prototype.initialize = function(cb) {
+  var that = this;
+  // Ensure that cb is called only once.
+  var pendingOperations = 2;
```

Comment:
`once` doesn't cover the exact desired behavior here, as we want the callback to be fired when the latter operation completes, not the former.